### PR TITLE
[SPARK-39334][BUILD] Exclude `slf4j-reload4j` from `hadoop-minikdc` test dependency

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
[HADOOP-18088 Replace log4j 1.x with reload4j](https://issues.apache.org/jira/browse/HADOOP-18088) , this pr adds the exclusion of `slf4j-reload4j` for `hadoop-minikdc` to clean up waring message about `Class path contains multiple SLF4J bindings` when run UTs.

### Why are the changes needed?
Cleanup `Class path contains multiple SLF4J bindings` waring when run UTs.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?

- Pass GA
- Manual test

for example, run `mvn clean test -pl core`

**Before** 

```
[INFO] Running test.org.apache.spark.Java8RDDAPISuite
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/Users/xxx/.m2/repository/org/apache/logging/log4j/log4j-slf4j-impl/2.17.2/log4j-slf4j-impl-2.17.2.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/Users/xxx/.m2/repository/org/slf4j/slf4j-reload4j/1.7.36/slf4j-reload4j-1.7.36.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.apache.logging.slf4j.Log4jLoggerFactory]
```

**After**

no above warnings

### CODE REVIEW

**Comment by LuciferYang** [comment]
cc @sunchao

**Comment by LuciferYang** [comment]
Looks like a change introduced by hadoop 3.3.3

https://github.com/apache/hadoop/compare/rel/release-3.3.2...rel/release-3.3.3

image

File: pom.xml [hunk 1]
```diff
@@ -1418,7 +1418,7 @@
           </exclusion>
           <exclusion>
             <groupId>org.slf4j</groupId>
-            <artifactId>slf4j-log4j12</artifactId>
+            <artifactId>slf4j-reload4j</artifactId>
```
Comment (wangyum) [inline pom.xml:1421]: Could we keep `slf4j-log4j12` and also add `slf4j-reload4j` to exclusions list?

Comment (LuciferYang) [reply pom.xml:1421]: ok ~ although Hadoop 3.3.3 no longer relies on `slf4j-log4j12 `

Comment (LuciferYang) [reply pom.xml:1425]: [be2e8cb](https://github.com/apache/spark/pull/36721/commits/be2e8cb8f55cc40b98722349da31fef698fb3c34)  fix this

**Comment by dongjoon-hyun** [comment]
cc @viirya

**Review by viirya — COMMENTED** [review]
Looks okay to me.

File: pom.xml (hunk 1 above)
Comment (wangyum) [reply pom.xml:1425]: User may build and test with Hadoop 3.3.2.

**Review by wangyum — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (LuciferYang) [reply pom.xml:1425]: > User may build and test with Hadoop 3.3.2 by specify the exact version of Hadoop.

Got it ~ thanks @wangyum

**Review by dongjoon-hyun — APPROVED** [review]
Could you update the following PR description according to the latest change, @LuciferYang ? We exclude both ones.
> this pr change to exclude slf4j-reload4j instead of slf4j-log4j12

For the code change, +1, LGTM.

**Review by viirya — APPROVED** [review]

**Comment by LuciferYang** [comment]
> Could you update the following PR description according to the latest change, @LuciferYang ? We exclude both ones.
> 
> > this pr change to exclude slf4j-reload4j instead of slf4j-log4j12
> 
> For the code change, +1, LGTM.

done

**Comment by LuciferYang** [comment]
thanks all

**Comment by sunchao** [comment]
LGTM too, thanks @LuciferYang !

**Pull request closed without merging**